rclcpp/test/rclcpp/test_publisher_with_type_adapter.cpp

```diff
@@ -166,17 +166,19 @@ TEST_F(TestPublisher, various_creation_signatures) {
 
 /*
  * Testing that conversion errors are passed up.
+ * TODO. This test was removed when implementing non convering intra-proccess
+ * publishing. It may be fixed and added again later.
```

Comment:
There is quite a good reason why we have to disable this test now, though it is slightly subtle.

Previous to this PR, anytime we used the TypeAdapt functionality in the rclcpp intra-process manager, we always converted immediately from a TypeAdapt type (i.e. `std::string`) to a ROS message type (i.e. `std_msgs::msg::String`).  Because of that, we always knew immediately if a TypeAdapted type was viable immediately on publish (regardless of whether there were subscribers), and hence this test made sense.

After this PR, we only convert to a ROS message type if one of the subscribers requires us to do that.  This is more efficient since we save a conversion to and from the custom type if the publisher and subscriber are of the same custom type and are using the intra-process manager to communicate.  However, this means that in the absence of any subscribers, we no longer attempt the conversion so this never throws.

There may be a way to keep this test with some additional fixes, but we haven't found a way to do that yet.  Another way to fix this is to add some subscribers of the ROS message type, which will cause the conversion to happen.